evalml/automl/engine/cf_engine.py

```diff
@@ -0,0 +1,140 @@
+from evalml.automl.engine.engine_base import (
+    EngineBase,
+    EngineComputation,
+    evaluate_pipeline,
+    score_pipeline,
+    train_pipeline,
+)
+
+
+class CFClient:
+    """Custom CFClient API to match Dask's CFClient and allow context management."""
+
+    def __init__(self, pool):
+        self.pool = pool
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, typ, value, traceback):
+        pass
+
+    def submit(self, *args, **kwargs):
+        """Pass through to imitate Dask's Client API."""
+        return self.pool.submit(*args, **kwargs)
+
+
+class CFComputation(EngineComputation):
+    """A Future-like wrapper around jobs created by the CFEngine.
+
+    Arguments:
+        future (callable): Computation to do.
+    """
+
+    def __init__(self, future):
+        self.work = future
+        self.meta_data = {}
+
+    def done(self):
+        """Whether the computation is done."""
+        return self.work.done()
+
+    def get_result(self):
+        """Gets the computation result.
+        Will block until the computation is finished.
+
+        Raises:
+             Exception: If computation fails. Returns traceback.
+        """
+        return self.work.result()
+
+    def cancel(self):
+        """Cancel the current computation."""
+        return self.work.cancel()
+
+    @property
+    def is_cancelled(self):
+        """Returns whether computation was cancelled."""
+        return self.work.cancelled()
+
+
+class CFEngine(EngineBase):
+    """The concurrent.futures (CF) engine"""
+
+    def __init__(self, client):
+        if not isinstance(client, CFClient):
+            raise TypeError(
+                f"Expected evalml.automl.engine.cf_engine.CFClient, received {type(client)}"
+            )
+        self.client = client
+        self._data_futures_cache = {}
+
+    def submit_evaluation_job(self, automl_config, pipeline, X, y) -> EngineComputation:
```

Comment:
nit-pick: Let's get rid of the type hint? I know the DaskEngine is guilty of this too so feel free to delete there as well hehe